eclipse-theia/theia
# Support remote port forwarding

#### What it does

Builds on top #13372. So that PR should probably be merged first

This adds the ability to the remote feature to forward ports to the remote system. Works for both ssh and dev-container connections.
Adds a new Port Widget to manage forwarded Ports.
![grafik](https://github.com/eclipse-theia/theia/assets/34068281/d01bf9e2-ecc3-4735-9ed7-69e0cfa37989)


#### How to test

1. Connect theia to a remote system either through ssh or the dev-container support
2. start some application on the remote system listening to a port. Like a simple node.js server for example
3. go to views->open-view...->Ports to open the Port forwarding UI
4. forward the port the application is listening to
5. connection to the application should work


#### Follow-ups



#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
